man/fi_cntr.3.md

```diff
@@ -121,7 +121,16 @@ struct fi_cntr_attr {
   and/or receives -- which are counted may be restricted using control
   flags when binding the counter and the endpoint.  Counters increment
   on all successful completions, separately from whether the operation
-  generates an entry in an event queue.
+  generates an entry in an event queue.  This includes truncated messages.
+
+- *FI_CNTR_EVENTS_BYTES*
+: The counter increments by the manipulated length for every successful 
+  completion that occurs on an associated bound endpoint.  The type of 
+  completions -- sends and/or receives -- which are counted may be 
+  restricted using control flags when binding the counter and the 
+  endpoint.  Counters increment on all successful completions, 
+  separately from whether the operation generates an entry in an 
+  event queue.  This includes truncated messages.
```

Comment:
It's unclear to me what the counter is actually counting.  I'm also unsure what is meant by 'manipulated length'.

The above definition references 'successful completions'.  Does that mean that transfers which do not generate a completion are not counted?  What about CQ data carried in a message?  Do those bytes get counted?  For atomics that return a result to the sender, is the result data also included at the initiator, and if so, how?  Is an atomic result counted a send data or receive data?  It may help to clarify exactly what gets counted for every type of data transfer which could be impacted by this feature.

There are a couple of ways an implementation can count the number of bytes of a data transfer.  One, it could increment the count as each packet is processed.  That would provide an accurate number of bytes which have been currently updated.  In the case of a failure, the count would still have been updated.  However, that behavior is not aligned with the above definition, which disables such an implementation.

If only fully successful transfers are updated, then the byte count must be tracked separately and updated only after the last packet has been processed.

I would not reference the term 'successful completion'.  The count is relative to either the initiator or target processing of the transfer.  The operation itself could fail.  For example, if the ack to the sender is lost, the receiver will still have the data and updated count, but the sender's operation may fail.